language/documentation/book/src/abort-and-assert.md

```diff
@@ -46,24 +55,28 @@ fun check_vec(v: &vector<u64>, bound: u64) {
 
 ### `assert`
 
-`assert` is a builtin, macro-like operation provided by the Move compiler. It takes two arguments, a condition of type `bool` and a code of type `u64`
+`assert` is a builtin, macro-like operation provided by the Move compiler. It takes two arguments, a
+condition of type `bool` and a code of type `u64`
 
 ```move
-assert(condition: bool, code: u64)
+assert!(condition: bool, code: u64)
 ```
 
-The operation does not exist at the bytecode level, and is replaced inside the compiler with
+Since the operation is a macro, it must be invoked with the `!`. This is to convey that the
+arguments to `assert` call-by-expression. In other words, `assert` is not a normal function and does
```

Comment:
```suggestion
arguments to `assert` are call-by-expression. In other words, `assert` is not a normal function and does
```
I'm not sure what "call-by-expression" means. I'm pretty sure it's not a verb. Maybe you could also explain a bit more?